AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/exceptions/ElementAlreadyExistsExceptionMapper.java

```diff
@@ -0,0 +1,16 @@
+package ch.mobi.itc.mobiliar.rest.exceptions;
+
+import ch.puzzle.itc.mobiliar.common.exception.ElementAlreadyExistsException;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+@Provider
+public class ElementAlreadyExistsExceptionMapper implements ExceptionMapper<ElementAlreadyExistsException> {
+    @Override
+    public Response toResponse(ElementAlreadyExistsException exception) {
+        return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(new ExceptionDto(exception)).build();
```

Comment:
Is there a better option than INTERNAL_SERVER_ERROR? The use won't now what went wrong! Maybe add an error message?

Probably one of these: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses

Maybe 409 Conflict

Reply:
Do I really need to add an error message? If so, what is the best way to do it? Because I think wherever this exception is thrown, a specific message is already set when the exception is created? Like 

```
if (getContextByName(newName) != null) {
    String message = "Der Kontext mit dem Namen " + newName + " ist bereits vorhanden und kann nicht erstellen werden";
   throw new ElementAlreadyExistsException(message, ContextEntity.class, newName);
            }
```
And when I create an ExceptionDTO with a Throwable ( `new ExceptionDto(exception)` ), the message of the Throwable is set in the ExceptionDTO:
```
public ExceptionDto(Throwable throwable) {		
	this.message = throwable.getMessage();
}
```

 So I think this particular message is better as if I would return some message. Or not?

Reply:
:+1: 
ok - no need for a specific message